json-rpc/types/src/views.rs

```diff
@@ -805,6 +837,36 @@ impl
     }
 }
 
+// TODO(philiphayes): is there some way to make these TryFrom while avoiding
+// circular deps? :(
+
+impl
+    TryInto<(
+        LedgerInfoWithSignatures,
+        EpochChangeProof,
+        AccumulatorConsistencyProof,
+    )> for &StateProofView
```

Comment:
Is there a reason this is done on a tuple type? Might be easier to just implement methods directly on StateProofView.

Reply:
It's definitely a bit cumbersome to shuttle these state proof tuples around, esp. given the long type names. Something I was considering was a new type `StateProof` that just contains the sub-proofs. It might make it a bit more convenient.

Reply:
Yeah I think that would be worth it

Reply:
I'll tackle this in a follow-up PR